### PR TITLE
usnic: fixed handling of mr_mode bits.

### LINKED ISSUE #3166: usnic: MR mode handling is a bit broken
If the `domain_attr->mr_mode` field is `0`, then we should fall back and check `hints->mode & FI_LOCAL_MR`. 

The `fi_rdm` test from fabtests currently sets `FI_LOCAL_MR` and leaves `domain_attr->mr_mode` empty. 

Unfortunately, this made it through the last manual `runfabtests.sh` run, @thananon need to investigate why it took Jenkins to find this. 

@thananon Please look into this.

### PR DESCRIPTION
**Pull request by thananon**
This commit is a fixed to previous PR to update to 1.5 contains bugs
where the provider reset given mode bits and caused undesired behavior.

This closes #3166 and #3165 .

Signed-off-by: Thananon Patinyasakdikul <apatinya@cisco.com>

### CODE REVIEW

**Comment by thananon** [comment]
Do not merge. I added something and something broke. Fixing.

**Comment by bturrubiates** [comment]
This should close both #3166 and #3165, right? If so, could you please add "This closes #3166, and closes #3165" to the PR body.

File: prov/usnic/src/usdf_dgram.h [hunk 1]
```diff
@@ -41,9 +41,8 @@
 
 #define USDF_DGRAM_CAPS (FI_MSG | FI_SOURCE | FI_SEND | FI_RECV)
 
-#define USDF_DGRAM_14_SUPP_MODE (FI_LOCAL_MR |FI_MSG_PREFIX)
+#define USDF_DGRAM_SUPP_MODE (FI_LOCAL_MR | FI_MSG_PREFIX)
 #define USDF_DGRAM_14_REQ_MODE (FI_LOCAL_MR)
```
Comment (bturrubiates) [inline prov/usnic/src/usdf_dgram.h]: Why did you remove 14 prefix from `SUPP_MODE` but not `REQ_MODE`?

Comment (thananon) [inline prov/usnic/src/usdf_dgram.h]: Because SUPP_MODE is the same for all version but only 1.4 required FI_LOCAL_MR.

**Pull request merged**